File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    var answer = true;
    let q =0;
    for(let i of s){
        if(i=="("){q++;}
        else{q--;}
        if(q<0){return false;}
    }
    if(q==0){return true;}
    else{return false;}
    return answer;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.Stack;

class Solution {
    boolean solution(String s) {
        Stack<Character> stack = new Stack<>();
        
        for (int i=0;i<s.length();i++) {
            char c = s.charAt(i);
            
            if(c == '(') {
                stack.push(c);
            }
            else if(c==')') {
                if(stack.isEmpty()) {
                    return false;
                }
                stack.pop();
            }
        }
            
            return stack.isEmpty();
        }

     
}